### PR TITLE
Missing whatsnew entries for 3.1 release.

### PR DESCRIPTION
**Pull request by pp-mo**
### Description
Missing and additional whatsnew content for the v3.1 release 

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: docs/src/whatsnew/latest.rst [hunk 1]
```diff
@@ -65,11 +65,8 @@ This document explains the changes made to Iris for this release
    :attr:`~iris.cube.Cube.attributes`, and improving spacing throughout.
    (:pull:`4206`)
 
-#. `@pp-mo`_ and `@lbdreyer`_ optimised loading netcdf files, resulting in a
-   speed up when loading with a single :func:`~iris.NameConstraint`. Note, this
-   optimisation only applies when matching on standard name, long name or
-   NetCDF variable name, not when matching on STASH.
-   (:pull:`4176`)
+#. `@schlunma`_ added support for loading atmosphere sigma coordinates from netcdf-CF files.  These now load as
```
Comment (pp-mo) [inline docs/src/whatsnew/latest.rst:68]: N.B. moved lhs content here to 'performance' section

File: docs/src/whatsnew/latest.rst [hunk 2]
```diff
@@ -122,6 +129,8 @@ This document explains the changes made to Iris for this release
 #. `@bjlittle`_ dropped both `black`_ and `flake8`_ package dependencies
    from our `conda`_ YAML and ``setup.cfg`` PyPI requirements. (:pull:`4181`)
 
+#. `@pp-mo`_ removed dependency on `PyKE_`. (:pull: `4198`)
```
Comment (pp-mo) [inline docs/src/whatsnew/latest.rst:132]: N.B. mentioned both here under 'dependencies', and in the 'internal' secition

**Comment by lbdreyer** [comment]
@pp-mo  at what point do you create rename the what'snew to 3.1.rst? is it after the 3.1.x branch is cut?

**Comment by pp-mo** [comment]
> @pp-mo at what point do you create rename the what'snew to 3.1.rst? is it after the 3.1.x branch is cut?

I would guess we do it in the 3.1.x branch.
From recent experience, we don't maintain a 'latest.rst' in the release/fix branches, only various "x.y.bug.rst" .
Though I personally still haven't quite grasped why (and it can make cross-merging between main + release branched a bit complicated).

So I expect to rename 'latest.rst' to '3.1.0.rst' within the release branch when we cut it.

**Review by lbdreyer — APPROVED** [review]

**Pull request merged**